show_client/ndp_cli.go

```diff
@@ -0,0 +1,418 @@
+		if !ok {
+			// sometimes it might be map[string]interface{}, so try that
+			if m, ok2 := val.(map[string]interface{}); ok2 {
+				attrs = make(map[string]string)
+				for k, v := range m {
+					attrs[k] = fmt.Sprintf("%v", v)
+				}
+			} else {
+				log.Warningf("Unexpected type for attrs: %T", val)
+				continue
+			}
+		}
+		// attrs is map[string]string
+		portIdRaw, ok := attrs["SAI_BRIDGE_PORT_ATTR_PORT_ID"]
+		if !ok {
+			continue
+		}
+		portId := portIdRaw[oidPrefix:] // strip "oid:0x"
+		// Map bridge port OID to port ID
+		ifBrOidMap[bridgePortOid] = portId
+	}
+	return ifBrOidMap, nil
+}
+
+func fetchFdbData() ([]BridgeMacEntry, error) {
+	queries := [][]string{
+		{"ASIC_DB", "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:*"},
+	}
+
+	// "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x2600000000063f\",\"mac\":\"B8:CE:F6:E5:50:05\",\"switch_id\":\"oid:0x21000000000000\"}"
+	brPortStr, err := GetMapFromQueries(queries)
+	if err != nil {
+		return nil, err
+	}
+	log.V(6).Infof("FDB_ENTRY list: %v", brPortStr)
+
+	ifOidMap, err := getInterfaceOidMap()
+	if err != nil {
+		return nil, err
+	}
+
+	ifBrOidMap, err := getBridgePortMap()
+	if err != nil {
+		return nil, err
+	}
+
+	if ifBrOidMap == nil || ifOidMap == nil {
+		return nil, fmt.Errorf("bridge/port maps not initialized")
+	}
+
+	bvidMap, err := buildBvidToVlanMap()
+	if err != nil {
+		log.Fatalf("Failed to build BVID map: %v", err)
+		return nil, err
+	}
+	oidPrefix := len("oid:0x")
+	bridgeMacList := []BridgeMacEntry{}
+
+	for fdbKey, entryData := range brPortStr {
+		// Split at first colon to separate top-level type from JSON
+		idx := strings.Index(fdbKey, ":")
+		if idx == -1 || idx+1 >= len(fdbKey) {
+			continue
+		}
+		fdbJSON := fdbKey[idx+1:] // everything after the first colon
+
+		fdb := map[string]string{}
+		if err := json.Unmarshal([]byte(fdbJSON), &fdb); err != nil {
+			continue
+		}
+
+		// Attributes map
+		ent, ok := entryData.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		brPortOidRaw, ok := ent["SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID"].(string)
+		if !ok || len(brPortOidRaw) <= oidPrefix {
+			continue
+		}
+		brPortOid := brPortOidRaw[oidPrefix:]
+
+		portID, ok := ifBrOidMap[brPortOid]
+		if !ok {
+			continue
+		}
+
+		ifName, ok := ifOidMap[portID]
+		if !ok {
+			ifName = portID
+		}
+
+		var vlanIDStr string
+		if v, ok := fdb["vlan"]; ok {
+			vlanIDStr = v
+		} else if bvid, ok := fdb["bvid"]; ok {
+			vlanIDStr, err = getVlanIDFromBvid(bvid, bvidMap)
+			if err != nil || vlanIDStr == "" {
+				vlanIDStr = bvid // fallback
+			}
+		} else {
+			continue
+		}
+
+		vlanID, err := strconv.Atoi(vlanIDStr)
+		if err != nil {
+			continue
+		}
+
+		bridgeMacList = append(bridgeMacList, BridgeMacEntry{
+			VlanID: vlanID,
+			Mac:    fdb["mac"],
+			IfName: ifName,
+		})
+	}
+
+	return bridgeMacList, nil
+}
+
+func getNDP(args sdc.CmdArgs, options sdc.OptionMap) ([]byte, error) {
+	intf, _ := options["interface"].String()
```

Comment:
Can you define a new option name, I believe we should be taking this as "iface" and not "interface"?